src/controllers/krate/owners.rs

```diff
@@ -170,8 +178,20 @@ async fn modify_owners(
                 "owners successfully removed".to_owned()
             };
 
-            Ok(Json(json!({ "ok": true, "msg": comma_sep_msg })))
-        })
+            Ok(comma_sep_msg)
+        })?;
+
+        // Send the accumulated invite emails now the database state has
+        // committed.
+        for email in emails {
+            let addr = email.recipient_email_address().to_string();
+
+            if let Err(e) = app.emails.send(&addr, email) {
+                warn!("Failed to send co-owner invite email: {e}");
+            }
+        }
+
+        Ok(Json(json!({ "ok": true, "msg": comma_sep_msg })))
```

Comment:
The controller collects these invite emails generated incrementally, and then sends them after the txn commits and the invites can no longer be rolled back.